src/ControllerDispatcher.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Zap\Routing;
+
+use Zap\Routing\Exceptions\InvalidRouteException;
+
+/**
+ * Class ControllerCaller
+ * @package Zap\Routing
+ * @author Gabor Zelei
+ */
+class ControllerDispatcher
+{
+    /**
+     * @var Route
+     */
+    private $route;
+
+    /**
+     * @var mixed       // @todo make sure this is actually Zap\Common\Interfaces\IController once IF is available
+     */
+    private $controller;
+
+    /**
+     * ControllerCaller constructor.
+     * @param Route $route
+     * @throws InvalidRouteException
+     */
+    public function __construct(Route $route)
+    {
+        $this->route = $route;
+
+        try {
+            $this->initController();
+        } catch (InvalidRouteException $e) {
+            throw $e;
+        }
+    }
+
+    /**
+     * Calls a controller based on current config
+     * @param array $params
+     * @return mixed
+     */
+    public function dispatchController(array $params = [])
```

Comment:
I'd rather go for something along the lines of what Laravel do, where they automatically generate a response from whatever the controller method returns.
E.g.: 
If return value is an `array` or an `object`, `Response` should render it into a `JSON` output.
In any other cases, the output should be a standard `string` for now. (We can refine it later on, if the need arises.)